state/application.go

```diff
@@ -369,12 +369,19 @@ func (op *DestroyApplicationOperation) destroyOps() ([]txn.Op, error) {
 	// If the application has no units, and all its known relations will be
 	// removed, the application can also be removed.
 	if op.app.doc.UnitCount == 0 && op.app.doc.RelationCount == removeCount {
-		hasLastRefs := bson.D{{"life", Alive}, {"unitcount", 0}, {"relationcount", removeCount}}
+		// If we're forcing destruction the assertion shouldn't be that
+		// life is alive, but that it's what we think it is now.
+		assertion := isAliveDoc
+		if op.Force {
+			assertion = bson.D{{"life", op.app.doc.Life}}
```

Comment:
I wonder if we need to assert on life when forcing at all... Maybe it should be just that the doc exists?

Reply:
Maybe. We can discuss with xtian and fix in all places this pattern is used.